src/Controllers/Core/MemberController.php

```diff
@@ -134,4 +141,65 @@ public function getLatestBoard(Request $request, Response $response, array $args
 
         return $response->withJson($latestBoard, 200);
     }
+
+    public function createProfilePicture(Request $request, Response $response, array $args)
+    {
+        $this->logger->debug("Uploading profile picture for member " . $args['id'] . " from " . $request->getServerParams()["REMOTE_ADDR"]);
+        
+        if ($request->getUploadedFiles() == null) {
+            $msg = 'No file given';
+            $this->logger->error($msg);
+            throw new KerosException($msg, 400);
+        }
+
+        $uploadedFile = $request->getUploadedFiles()['file'];
+        if ($uploadedFile == null) {
+            $msg = 'File is empty';
+            $this->logger->error($msg);
+            throw new KerosException($msg, 400);
+        }
+        if ($uploadedFile->getError() !== UPLOAD_ERR_OK) {
+            $msg = "Error during file uploading";
+            $this->logger->error($msg);
+            throw new KerosException($msg, 500);
+        }
+        $body = $args;
+        $body['file'] = $uploadedFile->getClientFileName();
+        
+        $this->entityManager->beginTransaction();
+        $filename = $this->memberService->createPhoto($args['id'], $body);
+        $this->entityManager->commit();
+
+        $filepath = $this->kerosConfig['MEMBER_PHOTO_DIRECTORY'] . $filename;
+
+        $uploadedFile->moveTo($filepath);
+
+        return $response->withStatus(204);
+    }
+
+    public function getProfilePicture(Request $request, Response $response, array $args)
+    {
+        $this->logger->debug("Getting profile picture for member " . $args['id'] . " from " . $request->getServerParams()["REMOTE_ADDR"]);
+
+        $filepath = $this->memberService->getPhoto($args["id"]);
+
+        $response   = $response->withHeader('Content-Type', 'application/image');

```

Comment:
I don't think this MIME type exists, does it ? I think we have to send "image/jpeg", or "image/png", or maybe easier for us "image/*" so we don't need to check ?

Reply:
Oof, sorry about that. It's fixed now. It now checks file extension before sending the photo.
I also added a file extension verification on upload ( and a test to go along with it )